src/app/transactions.component.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import TransactionsComponent from './transactions.component';
+import { By } from '@angular/platform-browser';
+import { ReactiveFormsModule } from '@angular/forms';
+
+describe('TransactionsComponent', () => {
+  let component: TransactionsComponent
+  let fixture: ComponentFixture<TransactionsComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [],
+      imports: [TransactionsComponent, ReactiveFormsModule]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TransactionsComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('Should Be Added In List on Button Click', () => {
+    const transactionsComponentsCount = component.transactions().length;
+    component.addTransaction({amount: 100, title: 'test'})
+    fixture.detectChanges();
+    expect(component.transactions().length).toBe(transactionsComponentsCount + 1);
+  });
+
+  it('Should Be Removed From List on Remove Btn Click', () => {
+    const transactionsComponentsCount = component.transactions().length;
+    const firstTransactionId = component.transactions()[0].id;
+    component.deleteTransaction(firstTransactionId)
+    fixture.detectChanges();
+    expect(component.transactions().length).toBe(transactionsComponentsCount - 1);
+  });
+
+  it('Should Open Add Transaction Dialog on Button Click', () => {
+    const addTransactionBtn = fixture.debugElement.query(By.css('#add-transaction-btn'));
+    addTransactionBtn.triggerEventHandler('click', null);
+    fixture.detectChanges();
+    expect(component.addTransactionDialogOpen()).toBe(true);
+  });
+
+  it('Current Balance Should Be Calculated Correctly', () => {
+    const transactions = component.transactions();
+    const currentBalance = transactions.reduce((acc, transaction) => {
```

Comment:
I would not use this code here because it is essentially the same code as in the component that you're testing. How do you know it is correct? Instead, you should've calculated it by hand and just put the amount in the expectation.